C#Backend/InpatientTherapySchedulingProgram/Models/Patient.cs

```diff
@@ -46,5 +46,48 @@ public Patient()
         public virtual ICollection<Appointment> Appointment { get; set; }
         [InverseProperty("Patient")]
         public virtual ICollection<PatientEvent> PatientEvent { get; set; }
+
+
+        public override bool Equals(object obj)
+        {
+            return this.Equals(obj as Patient);
+        }
+
+        public bool Equals(Patient patient)
+        {
+            if (patient is null)
+            {
+                return false;
+            }
+
+            if (Object.ReferenceEquals(this, patient))
+            {
+                return true;
+            }
+
+            return this.PatientId == patient.PatientId && this.FirstName == patient.FirstName && this.MiddleName == patient.MiddleName &&
```

Comment:
When comparing strings, I think we need to do string1.Equals(string2) to compare the values and not just the reference.